src/main/scala/org/scoverage/coveralls/CoverallsClient.scala

```diff
@@ -70,7 +70,7 @@ class ScalaJHttpClient extends HttpClient {
     val response = request.execute()
     CoverallHttpResponse(response.code, response.body)
   } catch {
-    case e: HttpException => CoverallHttpResponse(500, e.message)
+    case e: Exception => CoverallHttpResponse(500, e.getMessage)
```

Comment:
After some quick research, I believe there needs to be some handling of non-exception error cases using `response.isError` (see https://github.com/scalaj/scalaj-http/issues/71#issuecomment-116149418). 

Additionally, it would be apt to handle specific exceptions (such as `HttpStatusException`) rather than defaulting to all, especially considering you'd be able to actually provide the error code, rather than assuming it's a 500.

Reply:
Actually, I'm moderately sure your `CoverallHttpResponse(code, body) is fine for non-exception error cases!

The point about the exception handling is still pertinent though.